### PR TITLE
Added with_stem support

### PR DESCRIPTION
**Pull request by Gilthans**
#287

### CODE REVIEW

**Comment by pjbull** [comment]
Thanks @Gilthans, could you add:
 - [ ] test
 - [ ] update methods table (you can run [this script](https://github.com/drivendataorg/cloudpathlib/blob/master/docs/make_support_table.py))
 - [ ] update docs with note for `AnyPath` that some cloudpathlib methods may only be supported on compatible Python verions
 - [ ] update HISTORY.md

**Comment by pjbull** [comment]
Hi @Gilthans, would you have a chance to wrap what this PR needs? Should just be the [items above](https://github.com/drivendataorg/cloudpathlib/pull/290#issuecomment-1322713716) after rebasing onto the latest.

**Comment by Gilthans** [comment]
Sure, lost track of this PR completely!

**Comment by Gilthans** [comment]
> * [ ]  update docs with note for `AnyPath` that some cloudpathlib methods may only be supported on compatible Python verions
Where would you envision this appearing? In the `readme.md`? In the class docstring?

**Comment by Gilthans** [comment]
Another question: obviously tests are failing for python <3.9. Is there a way to specify that this test should only run on certain python versions?

File: cloudpathlib/cloudpath.py [hunk 1]
```diff
@@ -759,6 +759,9 @@ def suffix(self) -> str:
     def suffixes(self) -> List[str]:
         return self._dispatch_to_path("suffixes")
 
+    def with_stem(self, stem: str) -> Self:
+        return self._dispatch_to_path("with_stem", stem)
```
Comment (pjbull) [inline cloudpathlib/cloudpath.py:763]: CPython implementation is trivial, so maybe we put this in a try-except and fallback to our backport of their implementation if it is not present:
https://github.com/python/cpython/pull/19295/files#diff-fa525485738fc33d05b06c159172ff1f319c26e88d8c6bb39f7dbaae4dc4105cR859-R862

**Review by pjbull — CHANGES_REQUESTED** [review]

File: cloudpathlib/cloudpath.py (hunk 1 above)
Comment (pjbull) [reply cloudpathlib/cloudpath.py:763]: (this means that we won't need to skip the test or make a note about compatible versions)

**Review by pjbull — APPROVED** [review]

**Pull request merged**